### PR TITLE
SDPA Optimizations: Add mask fusion 

### PR DESCRIPTION
**Pull request by alingTT**
### Problem description
This PR does the following optimizations:

We fuses the attention mask addition within the matmul. The output of the matmul is produced in the DST, we can directly use `acc_to_dst` feature of `add_tiiles` to directly add the mask on top, hence we do not need to pack results back to L1 and load it back to DST for adding mask. 

### Checklist
- [x] 4u [Pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/16701913587)
- [x] 6u [Pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/16701963601)

### CODE REVIEW

**Comment by cglagovichTT** [comment]
Very cool! What is the impact on SDPA decode perf?

**Comment by alingTT** [comment]
> Very cool! What is the impact on SDPA decode perf?

In terms of kernel duration, there is about 0.6~0.7us savings (7% improvement).
On 4u:
Before mask fusion:
{
    "AVG DEVICE KERNEL SAMPLES/S": 3491416.933372126,
    "MIN DEVICE KERNEL SAMPLES/S": 3485459.1003158693,
    "MAX DEVICE KERNEL SAMPLES/S": 3502243.6248221514,
    "AVG DEVICE KERNEL DURATION [ns]": 9165.333333333334,
    "MIN DEVICE KERNEL DURATION [ns]": 9137.0,
    "MAX DEVICE KERNEL DURATION [ns]": 9181.0
}

After mask fusion:
{
    "AVG DEVICE KERNEL SAMPLES/S": 3770768.6869083624,
    "MIN DEVICE KERNEL SAMPLES/S": 3744003.7440037434,
    "MAX DEVICE KERNEL SAMPLES/S": 3823178.0167264035,
    "AVG DEVICE KERNEL DURATION [ns]": 8486.333333333334,
    "MIN DEVICE KERNEL DURATION [ns]": 8370.0,
    "MAX DEVICE KERNEL DURATION [ns]": 8547.0
}

On 6u:
Before Mask Fusion:
{
    "AVG DEVICE KERNEL SAMPLES/S": 3509028.4377512974,
    "MIN DEVICE KERNEL SAMPLES/S": 3482803.6569438395,
    "MAX DEVICE KERNEL SAMPLES/S": 3527336.8606701936,
    "AVG DEVICE KERNEL DURATION [ns]": 9119.333333333334,
    "MIN DEVICE KERNEL DURATION [ns]": 9072.0,
    "MAX DEVICE KERNEL DURATION [ns]": 9188.0
}
After Mask Fusion:
{
  "AVG DEVICE KERNEL SAMPLES/S": 3804240.1426590057,
  "MIN DEVICE KERNEL SAMPLES/S": 3797318.1440607565,
  "MAX DEVICE KERNEL SAMPLES/S": 3808616.9959533443,
  "AVG DEVICE KERNEL DURATION [ns]": 8411.666666666666,
  "MIN DEVICE KERNEL DURATION [ns]": 8402.0,
  "MAX DEVICE KERNEL DURATION [ns]": 8427.0
}

File: ttnn/cpp/ttnn/operations/transformer/sdpa_decode/device/kernels/compute/sdpa_flash_decode.cpp [hunk 1]
```diff
@@ -263,7 +271,26 @@ void MAIN {
                     qk_in0_block_w,
                     qk_subblock_h_dynamic,
                     qk_subblock_w_dynamic,
-                    true /*transpose*/);
+                    true,
+                    add_mask_fusion,
```
Comment (cglagovichTT) [inline ttnn/cpp/ttnn/operations/transformer/sdpa_decode/device/kernels/compute/sdpa_flash_decode.cpp:275]: I'd think that you need `add_mask_fusion && ((is_causal && (k_chunk == k_chunk_end - 1 && apply_mask_at_last_chunk)) || use_attention_attention_mask)` so that you only add mask when necessary

Comment (cglagovichTT) [reply ttnn/cpp/ttnn/operations/transformer/sdpa_decode/device/kernels/compute/sdpa_flash_decode.cpp:275]: I did not read 10 lines above!

**Review by cglagovichTT — APPROVED** [review]
